clang/include/clang/Lex/Lexer.h

```diff
@@ -582,6 +582,12 @@ class Lexer : public PreprocessorLexer {
   /// sequence.
   static bool isNewLineEscaped(const char *BufferStart, const char *Str);
 
+  /// Diagnose use of a delimited or named escape sequence.
+  static void DiagnoseDelimitedOrNamedEscapeSequence(SourceLocation Loc,
+                                                     bool Named,
+                                                     const LangOptions &Opts,
+                                                     DiagnosticsEngine &Diags);
+
```

Comment:
This could just be a non-member static function Lexer.cpp

Reply:
It's called from LiteralSupport.cpp, so it could be a non-member function, but I think it's a bit cleaner to scope it to the `Lexer` to make it clear this is emitting lexer-based diagnostics. WDYT?